src/convenience/session.ts

```diff
@@ -314,10 +310,24 @@ export function lazySession<S, C extends Context>(
     };
 }
 
+/** Stores session data per chat by default */
 function defaultGetSessionKey(ctx: Context): string | undefined {
     return ctx.chat?.id.toString();
 }
 
+/** Returns a useful error message for when the session key is undefined */
+function undef<C extends Context>(
+    op: "access" | "assign",
+    getSessionKey: SessionOptions<C>["getSessionKey"],
+    opts: { lazy?: boolean } = {},
```

Comment:
I like the named Boolean argument 👌

Reply:
Yeah I hate looking at a function signature and then there's this third argument _`true`_ … and I'm just sitting there, wondering what the hell that would be

Reply:
You probably feel the same :D

Reply:
The rust-analyzer provides inline hints so you can read the argument name of the variable.
It looks like `…, lazy: true)` while the actual source code is just `…, true)`. That helps a lot. But yeah, prefer enums (like `op` is done here) or named arguments, especially when they are optional. 👌